R/dp_accessors.R

```diff
@@ -203,6 +203,33 @@ get_women_on_art <- function(tag, tag_data, metadata, agegr_labels) {
   women_on_art
 }
 
+#' Get ART eligibility population data.
+#'
+#' @param tag The name of the tag getting data for.
+#' @param tag_data Tag data to be parsed.
+#' @param metadata Metadata params made available by tag configuration.
+#' @param agegr_labels Age group labels.
+#'
+#' @keywords internal
+get_eligibility_pop_data <- function(tag, tag_data, metadata, proj_years) {
+  if (is.null(metadata$rows) || is.null(metadata$cols)) {
+    stop(sprintfr("Can't get eligibility pop data via tag %s. Configuration is
+                  incomplete. Must specify rows and cols at minimum.
+                  rows are null: %s, cols are null: %s.",
+                  tag, is.null(metadata$rows), is.null(metadata$cols)))
+  }
+
+  data <- tag_data[metadata$rows, metadata$cols]
+  data <- stats::setNames(data, c("description", "pop", "elig", "percent", "year"))
+  data$pop <- c("PW", "TBHIV", "DC", "FSW", "MSM", "IDU", "OTHER")
+  data$elig <- as.logical(as.integer(data$elig))
+  data$percent <- as.numeric(data$percent)/100
+  data$year <- as.integer(data$year)
+  data$idx <- match(as.integer(data$year), proj_years)
+  rownames(data) <- data$pop
```

Comment:
why set rownames? is this depended on elsewhere?

Reply:
Having a quick look through `eppasm` and `first90` no I don't think the rowname is ever needed. Again this was just left in to preserve the naming structure that Jeff was using.

Reply:
rownames tend to get added pretty aggressively by data.frame construction functions (which is possibly how they ended up in the original data structure), but they behave weirdly in some manipulation operations, so I think omitting them is worthwhile if possible

Reply:
Okay noted that, I've taken them out.